tests/src/testUnified.cpp

```diff
@@ -86,47 +89,47 @@ boost::shared_ptr<NDRegisterAccessor<std::string>> ProcessArrayFactoryBackend::g
   if(path == "/unidir/sender") {
     flags.checkForUnknownFlags({}); // test expects that write-only accessors never accept wait_for_new_data...
     auto pv = createSynchronizedProcessArray<std::string>(1, path, "", "", "", 3, {AccessMode::wait_for_new_data});
-    _pv = pv.second;
+    _pv.push_back(pv.second);
     return pv.first;
   }
   if(path == "/unidir/polledSender") {
     flags.checkForUnknownFlags({}); // test expects that write-only accessors never accept wait_for_new_data...
     auto pv = createSynchronizedProcessArray<std::string>(1, path, "", "", "", 3, {});
-    _pv = pv.second;
+    _pv.push_back(pv.second);
     return pv.first;
   }
   if(path == "/unidir/receiver" && flags.has(AccessMode::wait_for_new_data)) {
     auto pv = createSynchronizedProcessArray<std::string>(1, path, "", "", "", 3, flags);
-    _pv = pv.first;
+    _pv.push_back(pv.first);
     // need to write initial value
-    _pv->accessData(0) = generateValueFromCounter();
-    _pv->write();
+    _pv.back()->accessData(0) = generateValueFromCounter();
```

Comment:
I don't how this worked before, but it sounds wrong. The UnifiedBackendTest must be able to get the value via getRemoteValue(), so it cannot be generated when you get an accessor. At least, if you get a second accessor, it should have the same initial value as the first one if the test does not call setRemoteValue().

Reply:
Looking at the implementation of getRemoteValue() and setRemoteValue() the implementation could be

```
std::string initialValue;
if (_pv[path].empty()) {
  initialValue = generateValueFromCounter();
}else{
  initialValue = _pv[path].back()->accessData(0);
}

auto pv = createSynchronizedProcessArray<std::string>(1, path, "", "", "", 3, flags);
_pv[path].push_back(pv.first);
_pv[path].back()->accessData(0) = initialValue;
```

Reply:
This is kind of a memory leak because you never clear out the unused, old pvs, but should be good enough for the lifetime of the test.

Reply:
I think the whole control system adapter has this conceptual "memory leak". I always keeps copies of all created pv pairs, so no harm done keeping one more pointer in the backend.

Reply:
changed